samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -261,6 +261,14 @@ protected void runStreamProcessor(SamzaResourceRequest request, String preferred
       throw new SamzaException("Expected resource for Processor ID: " + request.getProcessorId() + " was unavailable on host: " + preferredHost);
     }
 
+    if(clusterResourceManager.isResourceExpired(resource)) {
+      LOG.info("Lease for the resource has been expired for resource: {} request: {}", resource, request);
```

Comment:
Resource has expired, re-requesting resource for ..... processorId: 
Having the processorId: pattern in the logs makes for easy searcheability when we're debugging

Reply:
I came up with this PR to gather some early thoughts on approach, will clean it up